vortex-file/src/read/metadata.rs

```diff
@@ -33,74 +51,22 @@ impl<R: VortexReadAt + Unpin> MetadataFetcher<R> {
         root_layout: Box<dyn LayoutReader>,
         layout_cache: Arc<RwLock<LayoutMessageCache>>,
     ) -> Self {
-        Self {
+        let metadata_reader = BufferedLayoutReader::new(
             input,
             dispatcher,
-            root_layout,
+            stream::iter(iter::once(Ok(RowMask::new_valid_between(0, 1)))),
```

Comment:
why [0,1]

Reply:
This is bad abstraction. This rowmask is never used but I need to produce a stream of row masks to call the metadata read… need to think this one over